bundles/org.openhab.core.karaf/src/main/java/org/openhab/core/karaf/internal/FeatureInstaller.java

```diff
@@ -389,65 +392,90 @@ private synchronized void installFeatures(FeaturesService featuresService, Set<S
             featuresService.installFeatures(addons,
                     EnumSet.of(FeaturesService.Option.Upgrade, FeaturesService.Option.NoFailOnFeatureNotFound));
             logger.debug("Installed '{}'", StringUtils.join(addons, ", "));
-            for (String addon : addons) {
-                if (isInstalled(featuresService, addon)) {
-                    postInstalledEvent(addon);
+            try {
+                Feature[] features = featuresService.listInstalledFeatures();
+                for (String addon : addons) {
+                    if (isInstalled(features, addon)) {
+                        postInstalledEvent(addon);
+                    }
                 }
+            } catch (Exception e) {
+                logger.error("Failed retrieving features: {}", e.getMessage());
             }
         } catch (Exception e) {
             logger.error("Failed installing '{}': {}", StringUtils.join(addons, ", "), e.getMessage());
         }
     }
 
-    private static synchronized void installFeature(FeaturesService featuresService, String name) {
+    private synchronized void uninstallFeatures(FeaturesService featuresService, Set<String> addons) {
+        for (String addon : addons) {
+            uninstallFeature(featuresService, addon);
+        }
+    }
+
+    private static synchronized boolean installFeature(FeaturesService featuresService, String name) {
         try {
-            if (!isInstalled(featuresService, name)) {
+            Feature[] features = featuresService.listInstalledFeatures();
+            if (!isInstalled(features, name)) {
                 featuresService.installFeature(name);
-                if (isInstalled(featuresService, name)) {
+                features = featuresService.listInstalledFeatures();
+                if (isInstalled(features, name)) {
                     logger.debug("Installed '{}'", name);
                     postInstalledEvent(name);
+                    return true;
                 }
             }
         } catch (Exception e) {
             logger.error("Failed installing '{}': {}", name, e.getMessage());
         }
+        return false;
     }
 
-    private static synchronized void uninstallFeature(FeaturesService featuresService, String name) {
+    private static synchronized boolean uninstallFeature(FeaturesService featuresService, String name) {
         try {
-            if (isInstalled(featuresService, name)) {
+            Feature[] features = featuresService.listInstalledFeatures();
+            if (isInstalled(features, name)) {
                 featuresService.uninstallFeature(name);
                 logger.info("Uninstalled '{}'", name);
                 postUninstalledEvent(name);
+                return true;
             }
         } catch (Exception e) {
-            logger.error("Failed uninstalling '{}': {}", name, e.getMessage());
+            logger.debug("Failed uninstalling '{}': {}", name, e.getMessage());
```

Comment:
If you catch all `Exception` you are potentially also catching for example `NullPointerException`s and  they tend to miss a message, so you will be printing the message `null` which will make it troublesome to debug, in this case it is always better to log the complete stacktrace.